diff --git a/arch/arm/boot/dts/qcom/kenzo/kenzo-msm8956-mtp.dtsi b/arch/arm/boot/dts/qcom/kenzo/kenzo-msm8956-mtp.dtsi
index d66d1a8..d7967e8 100644
--- a/arch/arm/boot/dts/qcom/kenzo/kenzo-msm8956-mtp.dtsi
+++ b/arch/arm/boot/dts/qcom/kenzo/kenzo-msm8956-mtp.dtsi
@@ -190,17 +190,6 @@
 		vdd-supply = <&pm8950_l10>;
 	};
 
-	fpc_fpc1020 {
-		compatible = "fpc,fpc1020";
-		fpc,gpio_rst = <&msm_gpio 110 0>;
-		fpc,gpio_irq = <&msm_gpio 106 0>;
-		fpc,irq_flags = <0x2001>;
-		fpc,event-type = <4>;
-		fpc,event-code = <4>;
-		vcc_spi-supply = <&pm8950_l6>;
-		fpc,enable-wakeup;
-	};
-
 	goodix_fingerprint {
 		compatible = "goodix,fingerprint";
 		vcc_spi-supply  = <&pm8950_l10>;
diff --git a/arch/arm64/configs/lineageos_kenzo_defconfig b/arch/arm64/configs/lineageos_kenzo_defconfig
index 3316532..0a0a43e 100644
--- a/arch/arm64/configs/lineageos_kenzo_defconfig
+++ b/arch/arm64/configs/lineageos_kenzo_defconfig
@@ -1623,7 +1623,7 @@ CONFIG_WAKE_GESTURES=y
 CONFIG_INPUT_MISC=y
 # CONFIG_INPUT_AD714X is not set
 # CONFIG_INPUT_BMA150 is not set
-CONFIG_INPUT_FPC1020=y
+CONFIG_INPUT_FPC1020=n
 # CONFIG_INPUT_HBTP_INPUT is not set
 # CONFIG_INPUT_MMA8450 is not set
 # CONFIG_INPUT_MPU3050 is not set
diff --git a/drivers/input/misc/goodix/gf_spi.c b/drivers/input/misc/goodix/gf_spi.c
index 3274ea7..1820387 100644
--- a/drivers/input/misc/goodix/gf_spi.c
+++ b/drivers/input/misc/goodix/gf_spi.c
@@ -100,10 +100,6 @@ static LIST_HEAD(device_list);
 static DEFINE_MUTEX(device_list_lock);
 static struct gf_dev gf;
 
-#ifdef CONFIG_MACH_XIAOMI_KENZO
-extern int kenzo_fpsensor;
-#endif
-
 static int driver_init_partial(struct gf_dev *gf_dev);
 
 
@@ -738,13 +734,6 @@ static int gf_probe(struct platform_device *pdev)
 #endif
 	FUNC_ENTRY();
 
-#ifdef CONFIG_MACH_XIAOMI_KENZO
-	if (kenzo_fpsensor != 2) {
-		pr_err("board no gdx fpsensor\n");
-		return -ENODEV;
-	}
-#endif
-
 	/* Initialize the driver data */
 	INIT_LIST_HEAD(&gf_dev->device_entry);
 #if defined(USE_SPI_BUS)
